lib/note-list/index.tsx

```diff
@@ -375,15 +417,29 @@ export class NoteList extends Component<Props> {
   }
 
   handleShortcut = event => {
+    const { notes } = this.props;
     const { ctrlKey, key, metaKey, shiftKey } = event;
+    const {
+      selected: { index },
+    } = this.state;
+
+    const highlightedIndex = this.getHighlightedIndex(this.props);
 
     const cmdOrCtrl = ctrlKey || metaKey;
     if (
       cmdOrCtrl &&
       shiftKey &&
       (key === 'ArrowUp' || key.toLowerCase() === 'k')
     ) {
-      this.props.onSelectNote(this.props.nextNote.id);
+      if (-1 === highlightedIndex) {
+        return true;
+      }
+
+      if (index < 0 || !notes[index - 1]?.id) {
+        return true;
+      }
+
+      this.props.onSelectNote(notes[index - 1].id);
```

Comment:
we can also rely on that `?.` operator and collapse this check into an assertion…

```ts
const noteIdAbove = notes[index - 1]?.id;

if (noteIdAbove) {
  this.props.onSelectNote( noteIdAbove );
}
```